### PR TITLE
CORE-38788 - Update the edgebasepath based on the environment. 

### PR DESCRIPTION
**Pull request by shammowla**
Support test execution against different edge environments. Also added debug mode and watching Testcafe during execution.

## Description

1) Based on the edgeBasePath environment path passed from the test specs will execute against the given environment. 

2) Execute a single functional test spec instead of the entire suite by passing the "npm run ... test=specId".

3) Watch and debug using a "--watch" flag. 

## Motivation and Context

Test multiple against different environments. Write and debug tests more conveniently.  

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [ ] I have submitted a [documentation](https://github.com/AdobeDocs/alloy-docs) pull request or no changes are needed.
- [ ] I have made any necessary test changes and all tests pass.
- [ ] I have run the Sandbox successfully.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,8 +19,10 @@
     "test:server": "http-server ./ -a localhost -p 8080 -c-1",
     "testcafe": "testcafe",
     "functional:ci": "run-p --race test:server testcafe",
+    "functional:ci:int": "EDGE_ENV=ee-int run-p --race test:server testcafe",
```
Comment (jfkhoury) [inline package.json]: `EDGE_ENV=ee-int` needs updating as well: `EDGE_ENV=\"'ee-int'\"`

**Comment by shammowla** [comment]
@jfkhoury need one more approval to merge this. Added @ninaceban

File: package.json [hunk 2]
```diff
@@ -17,10 +17,12 @@
     "test:watch": "karma start",
     "test:watch:chrome": "karma start --browsers Chrome",
     "test:server": "http-server ./ -a localhost -p 8080 -c-1",
-    "testcafe": "testcafe",
+    "testcafe": "EDGE_ENV=\"'ee'\" testcafe",
```
Comment (liljenback) [inline package.json]: This is the environment but should we call this EDGE_BASE_PATH instead to be more consistent?

Comment (jfkhoury) [reply package.json]: @shammowla Let's merge, and add a TODO to maybe rename this VAR. Thanks @liljenback

**Comment by jfkhoury** [comment]
@owenleej, please look into this PR to see the new changes to the Functional Tests.

**Pull request merged**